src/main/java/org/folio/edge/rtac/RtacHandler.java

```diff
@@ -39,76 +41,207 @@
   private static final String PARAM_INSTANCE_ID = "instanceId";
   private static final String PARAM_INSTANCE_IDS = "instanceIds";
 
+  private static final String SUPPORTED_MIME_TYPE_PATTERN = "(text|\\*)\\s*/\\s*(xml|\\*)|(application|\\*)\\s*/\\s*(xml|json|\\*)";
+  private static final String XML_RELATED_MIME_TYPE_PATTERN = "(text|\\*)\\s*/\\s*(xml|\\*)|(application|\\*)\\s*/\\s*(xml|\\*)";
+  private static final String JSON_RELATED_MIME_TYPE_PATTERN = "(application)\\s*/\\s*(json)";
+  private static final String JSON_MIME_TYPE = "application/json";
+
   public RtacHandler(SecureStore secureStore, RtacOkapiClientFactory ocf) {
     super(secureStore, ocf);
   }
 
   protected void handle(RoutingContext ctx, boolean isBatch) {
     final var request = ctx.request();
-    logger.info("Request: {} \n params: {}", request.uri(), request.params() );
+    logger.info("Request: {} \n params: {}", request.uri(), request.params());
+
+    // Check if there were supported "Accept" headers passed
+    if (!checkSupportedAcceptHeaders(request)) {
+      notAcceptable(ctx, "Unsupported media type" + request.headers());
```

Comment:
Should it be?
```suggestion
      notAcceptable(ctx, "Unsupported media type" + request.headers().get("Accept"));
```

Reply:
Yep, it make sense to use particular header here, thanks.